### PR TITLE
Merge raj new view group

### PR DESCRIPTION
**Pull request by rgdixit**

### CODE REVIEW

File: app/src/main/java/loconsolutions/formwidgets/BaseBrokerSubForm.java [hunk 1]
```diff
@@ -83,22 +82,31 @@ private void initMandatoryCards(View view) {
...
+        }
+    }
+
+    private boolean isFirstTime() {
+        return BrokerRequestDashBoard.RES_UNOPENED == mFragState;
+    }
 
     private void addCards(BaseFormCard card) {
-        if (card.isCardMandatory())
+        if (card.isCardMandatory() && !(card instanceof CameraInputCard))
```
Comment (smmayank) [inline app/src/main/java/loconsolutions/formwidgets/BaseBrokerSubForm.java]: Do not add this check, instead handle case for CameraInputCard

File: app/src/main/java/loconsolutions/formwidgets/BaseBrokerSubForm.java [hunk 2]
```diff
@@ -110,8 +118,14 @@ protected final boolean isCardFilled() {
...
     }
 
+    protected void moveToSubForm(int formId) {
+        getActivity().onBackPressed();
+        mListener.moveToSubForm(formId);
+    }
+
     public interface OnBrokerCardFilledListener {
         void onBrokerCardFilled(int id, boolean isFilled);
+
```
Comment (smmayank) [inline app/src/main/java/loconsolutions/formwidgets/BaseBrokerSubForm.java]: Do not add unnecessary whitespace

File: app/src/main/java/loconsolutions/formwidgets/widgets/InputTextCard.java [hunk 3]
```diff
@@ -58,6 +53,7 @@ protected boolean isFilled() {
 
     public void registerOnTextChangedListener(OnTextChangedListener listener) {
         mListener = listener;
+        mInputText.addTextChangedListener(this);
```
Comment (smmayank) [inline app/src/main/java/loconsolutions/formwidgets/widgets/InputTextCard.java]: Wrong way of adding textListener, add text listener called multiple times, will lead to multiple call back

File: app/src/main/java/loconsolutions/formwidgets/widgets/InputTextCard.java [hunk 4]
```diff
@@ -66,18 +62,24 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
...
     }
 
     protected void postResult(String text) {
         if (null != mListener) {
             mListener.onTextChanged(this, text);
         }
+
+    }
+
+    protected  void postResultFromEditText(){
```
Comment (smmayank) [inline app/src/main/java/loconsolutions/formwidgets/widgets/InputTextCard.java]: do we need different, method

File: app/src/main/java/loconsolutions/formwidgets/widgets/PhoneNumberInputCard.java [hunk 5]
```diff
@@ -104,5 +105,56 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
             mCountryCodeEditText.setHint("CODE");
             mNumberEditText.setHint("Your LandLine Number");
         }
+        super.onCheckedChanged(buttonView, isChecked);
+    }
+
+    public void registerOnPhoneNumberInputListener(onPhoneNumberInputListener listener) {
+        mListener = listener;
+        mCountryCodeEditText.addTextChangedListener(this);
```
Comment (smmayank) [inline app/src/main/java/loconsolutions/formwidgets/widgets/PhoneNumberInputCard.java]: wrong way of adding listener

File: app/src/main/res/layout/fragment_broker_online_tech_details.xml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+                app:column_count="1"
+                app:header_text="What type of smartphone/OS do 
you use ?"
+                app:index_number="33"
+                app:mandatory="false"
+                app:multi_selection="true"
+                app:option_entries="@array/options_smartphones" />
+
+            <loconsolutions.formwidgets.widgets.OptionsCard
+            android:id="@+id/fragment_broker_online_tech_details_laptop"
```
Comment (smmayank) [inline app/src/main/res/layout/fragment_broker_online_tech_details.xml]: indentation?

File: app/src/main/java/loconsolutions/formwidgets/widgets/InputTextCard.java (hunk 4 above)
Comment (rgdixit) [inline app/src/main/java/loconsolutions/formwidgets/widgets/InputTextCard.java]: yup, we do for PhoneNumberInputCard to work smoothly
